gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposeCompilerKotlinSupportPlugin.kt

```diff
@@ -25,6 +26,23 @@ class ComposeCompilerKotlinSupportPlugin : KotlinCompilerPluginSupportPlugin {
                     ComposeCompilerCompatibility.compilerVersionFor(target.getKotlinPluginVersion())
             }
         }
+        warnAboutJetpackComposeCompilerUsageForNonJvm(target)
```

Comment:
Not sure yet, but may we should to call it inside `ComposeCompilerArtifactProvider`. We created Provider for a reason that on the `apply` stage isn't everything ready yet.

And if we do that, we need to show it only once, adding a check `isWarningShown`

Reply:
I moved the call to `project.afterEvaluate {}`, so it's called only once per project (gradle module with compose applied)